### PR TITLE
fugue optional dependency

### PR DESCRIPTION
**Pull request by fdosani**
Ref: #359 

- Initial PR to move Fugue as being an optional dependency.
- More explicit logging
- General cleanup and skipping of tests when module not found.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -11,7 +11,7 @@ maintainers = [
   { name="Faisal Dosani", email="faisal.dosani@capitalone.com" }
 ]
 license = {text = "Apache Software License"}
-dependencies = ["pandas<=2.2.3,>=0.25.0", "numpy<=2.1.3,>=1.22.0", "ordered-set<=4.1.0,>=4.0.2", "fugue<=0.9.1,>=0.8.7", "polars<=1.12.0,>=0.20.4"]
+dependencies = ["pandas<=2.2.3,>=0.25.0", "numpy<=2.1.3,>=1.22.0", "ordered-set<=4.1.0,>=4.0.2", "polars[pandas]<=1.12.0,>=0.20.4"]
```
Comment (ak-gupta) [inline pyproject.toml:14]: [non-blocking] do we need the `pandas` extra here? Based on their `pyproject.toml` it looks like the only addition is `pyarrow`

File: .github/workflows/test-package.yml [hunk 2]
```diff
@@ -134,7 +134,7 @@ jobs:
     - name: Install datacompy
       run: |
         python -m pip install --upgrade pip
-        python -m pip install .[tests,duckdb,polars,dask,ray]
+        python -m pip install .[tests,fugue]
```
Comment (ak-gupta) [inline .github/workflows/test-package.yml:137]: Should we have an environment for testing `datacompy` without `fugue` installed? Perhaps for an extra step of validation to make sure it's truly optional. Same for Snowflake and Spark I guess

File: pyproject.toml (hunk 1 above)
Comment (fdosani) [reply pyproject.toml:14]: Yeah there is some Pandas functionality which will fail without the `pandas` extras. I ran into this issue.

File: .github/workflows/test-package.yml (hunk 2 above)
Comment (ak-gupta) [reply .github/workflows/test-package.yml:137]: Oh yep, sorry didn't see that.

**Review by ak-gupta — APPROVED** [review]

File: datacompy/logger.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        The logging level. Defaults to ``logging.INFO``
+
+    Returns
+    -------
+    logging.Logger
+        Returns a Logger obejct which is set with the passed in paramters.
+        Please see the following for more details:
+        https://docs.python.org/2/library/logging.html
+    """
+    logging.basicConfig(format=FORMAT_TO_USE, stream=sys.stdout, level=log_level)
```
Comment (clrusby) [inline datacompy/logger.py:58]: It's not a great idea to change the logging format on the global `logging` package (or to set the `captureWarnings` flag).
These are settings which should be applied globally by the user codebase which is importing `datacompy`.
The net result of this change is that everybody's python logging will be set to log at debug level when they first `import datacompy` - which I'm sure is unexpected!
Suggest simply deleting these two lines.

**Pull request merged**